test/cocotb/osdtestlib/debug_interconnect.py

```diff
@@ -165,78 +244,75 @@ def read_register(self, dut, destination, source, reg_width, address,
                             access.
             address:        address of the register the value is to be read
                             from.
-            response:       buffer variable which the response of the dut will
-                            be stored in
+            rx_packet:      debug interconnect packet which will mirror what the
+                            dut replied with
 
         Returns:
-            -
+            response of the debug module of variable length in the rx_packet
 
         """
 
         # branch depending on the width of the accessed data word
-        if reg_width == 16:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_16
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16
-
-        elif reg_width == 32:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_32
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32
-
-        elif reg_width == 64:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_64
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64
-
-        elif reg_width == 128:
-
-            subtype = self.PACKET_SUBTYPE_REQ_READ_REG_128
-            expected_response = self.PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128
-
+        if word_width == 16:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_16
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_16
+        elif word_width == 32:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_32
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_32
+        elif word_width == 64:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_64
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_64
+        elif word_width == 128:
+            subtype = PACKET_SUBTYPE_REQ_READ_REG_128
+            expected_response = PACKET_SUBTYPE_RESP_READ_REG_SUCCESS_128
         else:
-            # display an error if an unsupported register width was chosen
-            raise TestFailure("Invalid register width parameter")
 
-        packet = DiPacket(destination = destination, source = source,
-                          packet_type = self.PACKET_TYPE_REG,
-                          packet_subtype = subtype, payload = address)
+            # display an error if an unsupported register width was chosen
+            raise RegAccessFailedException(message="An invalid register width\
+            parameter was chosen!")
 
+        tx_packet = DiPacket(destination=destination, source=source,
+                             packet_type=PACKET_TYPE_REG,
+                             packet_subtype=subtype, payload=address)
 
         # transmit the read request
-        yield self.send_packet(dut = dut, DiPacket = packet)
+        yield self.send_packet(dut=dut, packet=tx_packet)
+
+        response = []
+
+        # read the response
+        yield self.receive_packet(dut=dut, response=response)
```

Comment:
same as above: it's clear from the code that you're waiting for a response here.